src/Compilers/CSharp/Test/Semantic/Diagnostics/IInvocationExpressionTests.cs

```diff
@@ -0,0 +1,800 @@
+            Assert.Equal(argumentValue.Type.SpecialType, SpecialType.System_Int32);
+            Assert.True(argumentValue.ConstantValue.HasValue);
+            Assert.Equal(argumentValue.ConstantValue.Value, 1);
+
+            // b: 2
+
+            argument = arguments[1];
+            Assert.True(argument == evaluationOrderArguments[0]);
+            Assert.False(argument.IsInvalid);
+            Assert.Null(argument.InConversion);
+            Assert.Null(argument.OutConversion);
+            Assert.Equal(argument.Parameter.Name, "b");
+            Assert.True(invocation.GetArgumentMatchingParameter(argument.Parameter) == argument);
+            argumentValue = argument.Value;
+            Assert.Equal(argumentValue.Kind, OperationKind.LiteralExpression);
+            Assert.False(argumentValue.IsInvalid);
+            Assert.Equal(argumentValue.Type.SpecialType, SpecialType.System_Int32);
+            Assert.True(argumentValue.ConstantValue.HasValue);
+            Assert.Equal(argumentValue.ConstantValue.Value, 2);
+
+            // M3(x)
+
+            Assert.Equal("M3(x)", nodes[2].ToString());
+            operation = model.GetOperation(nodes[2]);
+            Assert.Equal(operation.Kind, OperationKind.InvocationExpression);
+            Assert.False(operation.IsInvalid);
+            invocation = (IInvocationExpression)operation;
+            Assert.Equal(invocation.Type.SpecialType, SpecialType.System_Double);
+            Assert.Equal(invocation.TargetMethod.Name, "M3");
+            arguments = invocation.ArgumentsInParameterOrder;
+            Assert.Equal(arguments.Length, 1);
+
+            evaluationOrderArguments = invocation.ArgumentsInEvaluationOrder;
+            Assert.Equal(evaluationOrderArguments.Length, 1);
+
+            // x
+
+            argument = arguments[0];
+            Assert.True(argument == evaluationOrderArguments[0]);
+            Assert.False(argument.IsInvalid);
+            Assert.Null(argument.InConversion);
+            Assert.Null(argument.OutConversion);
+            Assert.Equal(argument.Parameter.Name, "d");
+            Assert.True(invocation.GetArgumentMatchingParameter(argument.Parameter) == argument);
+            argumentValue = argument.Value;
+            Assert.Equal(argumentValue.Kind, OperationKind.ConversionExpression);
+            Assert.False(argumentValue.IsInvalid);
+            Assert.Equal(argumentValue.Type.SpecialType, SpecialType.System_Double);
+            IConversionExpression conversion = (IConversionExpression)argumentValue;
+            argumentValue = ((IConversionExpression)argumentValue).Operand;
+            Assert.Equal(argumentValue.Type.SpecialType, SpecialType.System_Int32);
+            Assert.Equal(argumentValue.Kind, OperationKind.LocalReferenceExpression);
+            Assert.Equal(((ILocalReferenceExpression)argumentValue).Local.Name, "x");
+        }
+
+        [Fact]
+        public void ParamArrayInvocations()
```

Comment:
Adding that test exposed a bug. I think the cause is reported elsewhere in this PR.